scipy/sparse/_coo.py

```diff
@@ -1275,6 +1262,460 @@ def _broadcast_to(self, new_shape, copy=False):
 
         return coo_array((new_data, new_coords), new_shape)
 
+    def _process_arrays_for_comparison(self, other, op_name):
+        if is_pydata_spmatrix(other):
+            return NotImplemented
+
+        if not (issparse(other) or isdense(other) or isscalarlike(other)):
+            # If it's a list or whatever, treat it like an array
+            other_a = np.asanyarray(other)
+
+            if other_a.ndim == 0 and other_a.dtype == np.object_:
+                return NotImplemented
+
+            try:
+                other.shape
+            except AttributeError:
+                other = other_a
+
+        if self.ndim < 3 and (isscalarlike(other) or other.ndim < 3):
+            return getattr(_data_matrix, op_name)(self.tocsr(), other)
```

Comment:
We shouldn't be calling `self.tocsr()` here, we can let the base class method handle conversion if needed.